astroquery/utils/commons.py

```diff
@@ -84,13 +84,16 @@ def send_request(url, data, timeout, request_type='POST', headers={},
         if request_type == 'GET':
             response = requests.get(url, params=data, timeout=timeout,
                                     headers=headers, **kwargs)
-            return response
         elif request_type == 'POST':
             response = requests.post(url, data=data, timeout=timeout,
                                      headers=headers, **kwargs)
-            return response
         else:
             raise ValueError("request_type must be either 'GET' or 'POST'.")
+
+        if response.status_code >= 400:
+            raise response.raise_for_status()
```

Comment:
I think this can just be `response.raise_for_status()`, you don't need the if statement or anything.  At least, I thought that's what `raise_for_status` was supposed to do (raise an exception if status is bad, else pass)